### PR TITLE
Make cloth by adding vertices and springs

### PR DESCRIPTION
**Pull request by r41k0u**

### CODE REVIEW

File: 2DClothSim.h [hunk 1]
```diff
@@ -5,5 +5,55 @@
...
+		std::vector<struct vect2D> *velos, *accs;
+		float springConst, dampConst, vertMass;
+
+		const struct vect2D dirX = vect2D(1.0f, .0f), dirY = vect2D(.0f, 1.0f), dirNE = vect2D(sqrtf(0.5f), -sqrtf(0.5f)), dirSE = vect2D(sqrtf(0.5f), sqrtf(0.5f));
+
+		void initVerts();
+
+	public:
+		// Constructor
+		Cloth(int x = 100, int y = 100) : pixels_x(x), pixels_y(y), init(false), verts(nullptr), velos(nullptr), springConst(1.0f), dampConst(0.2f), vertMass(1.0f) { initVerts(); }
```
Comment (sin3point14) [inline 2DClothSim.h]: didn't have a close look but 100 here and 10 at https://github.com/r41k0u/2DClothSim/pull/2/files#diff-18ad10c6fe7062603f10d7e8fa4f6b3dfa1e95feac928d9057ac4ca9f43364d4R48

Comment (r41k0u) [reply 2DClothSim.h]: (10, 10) is the origin of the cloth, where the top left corner is. 2 is the distance between vertices

File: 2DClothSim.cpp [hunk 2]
```diff
@@ -3,30 +3,99 @@
...
+
+void ClothSim::update(Cloth* cloth, const float time) {
+    cloth->updateVelo(time);
+    cloth->updatePos(time);
+}
+
+void ClothSim::Cloth::updateAcc(const float time) {
+    calcStruct(time);
+    calcBend(time);
+    calcBend(time);
```
Comment (Aviii06) [inline 2DClothSim.cpp:69]:
```suggestion
    calcShear(time);
```

File: 2DClothSim.cpp [hunk 3]
```diff
@@ -3,30 +3,99 @@
...
+
+    init = true;
+}
+
+void ClothSim::update(Cloth* cloth, const float time) {
+    cloth->updateVelo(time);
+    cloth->updatePos(time);
+}
+
+void ClothSim::Cloth::updateAcc(const float time) {
```
Comment (Aviii06) [inline 2DClothSim.cpp:61]:
```suggestion
void ClothSim::Cloth::updateAcc(const float delTime) {
```

File: 2DClothSim.cpp [hunk 4]
```diff
@@ -3,30 +3,99 @@
...
+            accs->push_back(vect2D(.0f, 9.8f));
+            (*verts)[(i * pixels_x) + j].x = (float)(10 + 2 * j);
+            (*verts)[(i * pixels_x) + j].y = (float)(10 + 2 * i);
+        }
+    }
+
+    init = true;
+}
+
+void ClothSim::update(Cloth* cloth, const float time) {
```
Comment (Aviii06) [inline 2DClothSim.cpp:56]: Where is acc getting updated???

File: 2DClothSim.cpp [hunk 5]
```diff
@@ -3,30 +3,99 @@
...
+    for (int i = 0; i < pixels_x * pixels_y; i++) {
+        (*verts)[i].x += (*velos)[i].x * time;
+        (*verts)[i].y += (*velos)[i].y * time;
+    }
+}
+
+void ClothSim::Cloth::calcStruct(const float time) {
+    for (int64_t i = 0; i < pixels_y; i++) {
+        for (int64_t j = 0; j < pixels_x; j++) {
+            if (isValidVert(j + 1, i)) {
```
Comment (Aviii06) [inline 2DClothSim.cpp:90]: can't you just do: `j=0; j < pixels_x - 1`

File: 2DClothSim.cpp [hunk 6]
```diff
@@ -3,30 +3,99 @@
...
+    verts = new std::vector<SDL_FPoint>;
+    velos = new std::vector<struct vect2D>;
+    accs = new std::vector<struct vect2D>;
+
+    for (int64_t i = 0; i < pixels_y; i++) {
+        for (int64_t j = 0; j < pixels_x; j++) {
+            verts->push_back(SDL_FPoint());
+            velos->push_back(vect2D(.0f, .0f));
+            accs->push_back(vect2D(.0f, 9.8f));
+            (*verts)[(i * pixels_x) + j].x = (float)(10 + 2 * j);
```
Comment (Aviii06) [inline 2DClothSim.cpp]: Why not just push_back this????

File: 2DClothSim.cpp [hunk 7]
```diff
@@ -3,30 +3,99 @@
...
+        (*verts)[i].x += (*velos)[i].x * time;
+        (*verts)[i].y += (*velos)[i].y * time;
+    }
+}
+
+void ClothSim::Cloth::calcStruct(const float time) {
+    for (int64_t i = 0; i < pixels_y; i++) {
+        for (int64_t j = 0; j < pixels_x; j++) {
+            if (isValidVert(j + 1, i)) {
+                float delL = norm((*verts)[pixels_x * i + j].x - (*verts)[pixels_x * i + j + 1].x, (*verts)[pixels_x * i + j].y - (*verts)[pixels_x * i + j + 1].y) - 1.0f;
```
Comment (Aviii06) [inline 2DClothSim.cpp]: this seems weird... I am assuming this is incomplete?

File: 2DClothSim.cpp (hunk 4 above)
Comment (r41k0u) [reply 2DClothSim.cpp:56]: not yet

File: 2DClothSim.cpp (hunk 5 above)
Comment (r41k0u) [reply 2DClothSim.cpp:90]: No, I plan to check all 4 nearest neighbours for every vertex

File: 2DClothSim.cpp (hunk 6 above)
Comment (r41k0u) [reply 2DClothSim.cpp]: `SDL_FPoint` does not have such a constructor. I need to create an empty point and then populate its coordinates

File: 2DClothSim.cpp (hunk 7 above)
Comment (r41k0u) [reply 2DClothSim.cpp]: yes

File: 2DClothSim.cpp (hunk 5 above)
Comment (Aviii06) [reply 2DClothSim.cpp:90]: I see, though I think you can iterate twice once horizontally and once vertically. reverse iteration is not needed cause delL would be identical. i.e.only need to check for two neighbours.

**Pull request merged**